ddl_atomicity/ddl_atomicity_check_script_python2.py

```diff
@@ -0,0 +1,175 @@
+# -*- coding: UTF8 -*-
+
+from collections import defaultdict
+import HTMLParser
+import json
+import os
+import sys
+import subprocess
+import argparse
+
+# Define command-line arguments
+parser = argparse.ArgumentParser(description='Script to validate YSQL tables and columns.')
+parser.add_argument('--ysqlsh_path', type=str, default="/home/yugabyte/tserver/bin/ysqlsh", help='Path to ysqlsh executable.')
+parser.add_argument('--master_conf_path', type=str, default='/home/yugabyte/master/conf/server.conf', help='Path to master configuration file.')
+parser.add_argument('--master_interface_address', type=str, default=None, help='Master UI interface IP. If not provided, will be read from master_conf_path.')
+parser.add_argument('--master_interface_port', type=int, default=7000, help='Port for the master UI interface.')
+parser.add_argument('--ysql_host', type=str, default="/tmp/.yb.0.0.0.0:5433/", help='Host for ysqlsh.')
+parser.add_argument('--master_leader_only', action='store_true', help='Check if the node is the master leader and exit if it is not.')
+args = parser.parse_args()
+
+# Initialize error count and message list
+errorcount = 0
+error_messages = []
+
+FNULL = open(os.devnull, 'w')
+html_parser = HTMLParser.HTMLParser()
+
+# Check if the current node is the master leader if the flag is set
+if args.master_leader_only:
+    print("Checking if the node is the master leader.")
+    is_leader = subprocess.check_output(
+        r"curl -s http://localhost:9300/metrics | grep yb_node_is_master_leader\{ | awk '{print $2}'", 
```

Comment:
The paths to `curl`, `grep`, `awk`, etc. should be set as variables or called using `command -p <binary>` (assuming Python's `subprocess` module is shelling out to `bash`, anyway) to ensure the script doesn't break if PATH is unset or incorrect.

Reply:
Ah I misunderstood your previous comment, thanks for clarifying, will update